lib/deep-required/index.ts

```diff
@@ -18,6 +20,12 @@ export type DeepRequired<Type> = Type extends Error
   ? WeakSet<DeepRequired<Values>>
   : Type extends Promise<infer Value>
   ? Promise<DeepRequired<Value>>
+  : Type extends AnyArray<infer Values>
+  ? Type extends Tuple
```

Comment:
As I suggested above, you can use `IsNever<IsTuple<Type>>` so you will filter out tuple at this point so you don't have to duplicate the same code for objects and tuples

Reply:
Your suggestion isn't totally clear to me, so I opted instead to follow the pattern in deep-partial and duplicate the object code.